src/Stratis.Bitcoin.Features.PoA/Voting/VotingData.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Features.PoA.Voting
+{
+    /// <summary>
+    /// Type of vote that determines actions that will take place in case
+    /// majority of federation members votes in favor.
+    /// </summary>
+    public enum VoteKey : ushort
+    {
+        /// <summary>Remove one of the federation members.</summary>
+        KickFederationMember = 0,
+
+        /// <summary>Add new federation member.</summary>
+        AddFederationMember = 1
+    }
+
+    /// <summary>Information about a single vote.</summary>
+    public class VotingData : IBitcoinSerializable
+    {
+        public VoteKey Key
+        {
+            get => (VoteKey)this.key;
+            set => this.key = (ushort)value;
+        }
+
+        private ushort key;
+
+        public byte[] Data;
+
+        public static bool operator ==(VotingData a, VotingData b)
+        {
+            if (ReferenceEquals(a, b))
+                return true;
+
+            if (((object)a == null) || ((object)b == null))
+                return false;
+
+            if (((object)a.Data == null) || ((object)b.Data == null))
+                return false;
+
+            return a.Key == b.Key && a.Data.SequenceEqual(b.Data);
+        }
+
+        public static bool operator !=(VotingData a, VotingData b)
+        {
+            return !(a == b);
+        }
+
+        public override bool Equals(object obj)
+        {
+            VotingData item = obj as VotingData;
+
+            if (item == null)
+            {
+                return false;
+            }
+
+            return this == item;
+        }
+
+        public void ReadWrite(BitcoinStream stream)
+        {
+            stream.ReadWrite(ref this.key);
+
+            if (stream.Serializing)
+            {
+                ushort dataSize = (ushort)this.Data.Length;
+
+                stream.ReadWrite(ref dataSize);
```

Comment:
Don't you get this functionality out of the box with the serializer?

Reply:
hmm yeah there is a trick there I forgot what it was, look at trx class for OutputList or something for an example, however I don't mind if you just prefer to do it manually.